### PR TITLE
[New Rule] Potential Cookies Theft via Browser Debugging

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/740

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+process where event.type in ("start", "process_started", "info") and
+  process.name in (
+             "chrome.exe",
+             "Google Chrome",
+             "google-chrome-stable",
+             "google-chrome-beta",
+             "google-chrome",
+             "msedge.exe") and
+   wildcard(process.args, "--remote-debugging-port=*")
+'''
```
Comment (brokensound77) [inline rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml:42]:
```suggestion
query = '''
process where event.type in ("start", "process_started", "info") and
  process.name in (
             "chrome.exe",
             "Google Chrome",
             "google-chrome-stable",
             "google-chrome-beta",
             "google-chrome",
             "msedge.exe") and
   process.args : "--remote-debugging-port=*" and
   process.args : "--user-data-dir=*"
'''
```

File: rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+maturity = "production"
+updated_date = "2020/12/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the execution of a Chromium based browser with the debugging process argument, this may indicate an attempt
+to steal authentication cookies. An adversary may steal web application or service session cookies and use them to gain
+access web applications or Internet services as an authenticated user without needing credentials.
+"""
```
Comment (brokensound77) [inline rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml:12]:
```suggestion
Identifies the execution of a Chromium based browser with the debugging process argument, which may indicate an attempt
to steal authentication cookies. An adversary may steal web application or service session cookies and use them to gain
access web applications or Internet services as an authenticated user without needing credentials.
"""
```

File: rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+false_positives = ["Developers performing browsers plugin or extension debugging."]
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
```
Comment (brokensound77) [inline rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml:15]: could this work on auditbeat too?

**Review by brokensound77 — COMMENTED** [review]
Nice rule, thanks 👍

File: rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml (hunk 1 above)
Comment (brokensound77) [reply rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml:42]: looks like `--remote-debugging-targets` and `--remote-debugging-pipe` may be able to [achieve this too](https://peter.sh/experiments/chromium-command-line-switches/). So maybe extend it to

```suggestion
query = '''
query = '''
process where event.type in ("start", "process_started", "info") and
  process.name in (
             "chrome.exe",
             "Google Chrome",
             "google-chrome-stable",
             "google-chrome-beta",
             "google-chrome",
             "msedge.exe") and
   process.args : ("--remote-debugging-port=*", "--remote-debugging-targets=*",  "--remote-debugging-pipe=*") and
   process.args : "--user-data-dir=*"
'''```

File: rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml (hunk 3 above)
Comment (Samirbous) [reply rules/cross-platform/credential_access_cookies_chromium_browsers_debugging.toml]: good catch

**Review by brokensound77 — APPROVED** [review]
👍

**Review by threat-punter — APPROVED** [review]
👍

**Pull request merged**